APIs/AuthorizationAPI.raml

```diff
@@ -4,8 +4,8 @@
 # (c) AMWA 2020
 
 title: NMOS Authorization
-baseUri: http://api.example.com/{issuer}
-issuer: x-nmos/auth/v1.0
+baseUri: https://api.example.com/{version}
+version: x-nmos/auth/v1.0
```

Comment:
Is this intended to indicate that commercial auth servers don't have to use the 'x-nmos' pattern? If so could we use a name like 'optionalNMOSPath' perhaps? Version seems a little confusing given it contains more than just that.

Reply:
`raml2html` didn't seem to build when it was named issuer. Changing it to version appeared to fix it.

Reply:
I'm not sure if 'issuer' would have been any clearer, unless I'm missing something?

Reply:
I agree neither seem very good. `basePath`? Trouble is we're needing to be more flexible here than we have elsewhere. There's probably some interaction with the DNS-SD TXT records too, which elsewhere can always be used to build the full URL.

Reply:
... in which case maybe `selector`? Although, if the `api_selector` TXT record includes the `x-nmos/auth/{version}`, I'm concerned that we now actually mean something different by selector in IS-10 than in IS-08?

Reply:
I'm not sure that is what this is. The 'issuer', which corresponds to the 'api_selector' is only relevant for the OAuth metadata RFC, which is covered by the other RAML document (RFC 8414 section 3.1). This ought to use a completely different name to avoid being confused with those.

Reply:
Ah, sorry if I got the wrong end of the stick!

Reply:
We've just had a chat about this as it's thoroughly confusing and not entirely helped by the language in the RFC. It does appear that 'issuer' is roughly the right name to use here as (if present) the 'issuer' path must match between the main Authorization Server and its metadata endpoint. The confusing element is that the 'issuer' in the metadata includes a protocol and hostname too, so this variable in the RAML will be changed to 'issuerIDPath' or similar to attempt to be less ambiguous.

A further confusing thing is that if you wish to write your own Auth server and you happen to want to conform to the general x-nmos path convention, you would have to set your 'api_selector' to 'x-nmos/auth/v1.0' in order to cause it to become the issuer ID path. This makes the API paths look familiar, but makes it look like the 'api_selector' is being mis-used.

Reply:
Yep, that's confusing.